backend/tests/api/banner/index.js

```diff
@@ -0,0 +1,166 @@
+const request = require('supertest');
+const { jwt } = require("../../helpers/strapi");
+const userFactory = require("../../user/factory");
+
+describe("Banner Test", () => {
+    let authenticatedUser;
+    let adminUser;
+    beforeAll(async () => {
+        authenticatedUser = await userFactory.createUser(strapi, "authenticated");
+        adminUser = await userFactory.createUser(strapi, "admin");
+    });
+
+    /**
+     * @param {string | number} id
+     */
+    const constructBanner = (id) => {
+        return {
+            data: {
+                "Title": "Banner Title " + id,
+                "Description": "Banner Description " + id,
+                "Link": {
+                    "id": 1,
+                    "Label": "NY KEA",
+                    "URL": "www.nykea.org"
+                },
+                "StartDate": "2024-01-26",
+                "EndDate": "2024-02-26",
+                "locale": "en"
+            }
+        };
+    }
+
+    it("Public user should not post banner", async () => {
+        await request(strapi.server.httpServer)
+            .post("/api/banners")
+            .set("accept", "application/json")
+            .send(constructBanner(0))
+            .expect("Content-Type", /json/)
+            .expect(500)
+    })
+
+    it("Authenticated user should not post banner", async () => {
+        await request(strapi.server.httpServer)
+            .post("/api/banners")
+            .set("accept", "application/json")
+            .set("Authorization", `Bearer ${await jwt(authenticatedUser.id)}`)
+            .send(constructBanner(0))
+            .expect("Content-Type", /json/)
+            .expect(500)
+    })
+
+    it("Admin user should post banners", async () => {
+        await request(strapi.server.httpServer)
+            .post("/api/banners")
+            .set("accept", "application/json")
+            .set("Authorization", `Bearer ${await jwt(adminUser.id)}`)
+            .send(constructBanner(1))
+            .expect("Content-Type", /json/)
+            .expect(200)
+
+        await request(strapi.server.httpServer)
+            .post("/api/banners")
+            .set("accept", "application/json")
+            .set("Authorization", `Bearer ${await jwt(adminUser.id)}`)
+            .send(constructBanner(2))
+            .expect("Content-Type", /json/)
+            .expect(200)
+    })
+
+    it("Public user should find banners", async () => {
+        await request(strapi.server.httpServer)
+            .get("/api/banners?populate=*")
```

Comment:
A Banner will have Link as an attribute and Link is a separate table. When you get Banner without the pupulate option, it will populate only specific relations one-level deep like this
```
{
      "id": 1,
      "attributes": {
        "Title": "Banner Title",
        "Description": "Banner Description",
        "StartDate": "2024-02-08",
        "EndDate": null,
        "createdAt": "2024-02-08T23:24:46.009Z",
        "updatedAt": "2024-02-08T23:24:47.891Z",
        "publishedAt": "2024-02-08T23:24:47.887Z",
        "locale": "en"
      }
}
```
If you do populate=* It means it's go until the deepest level. So the result will be 
```
{
      "id": 1,
      "attributes": {
        "Title": "Banner Title",
        "Description": "Banner Description",
        "StartDate": "2024-02-08",
        "EndDate": null,
        "createdAt": "2024-02-08T23:24:46.009Z",
        "updatedAt": "2024-02-08T23:24:47.891Z",
        "publishedAt": "2024-02-08T23:24:47.887Z",
        "locale": "en",
        "Link": {
          "id": 3,
          "Label": "Button Text",
          "URL": "Button URL"
        },
        "localizations": {
          "data": []
        }
      }
    }
```

For more information, refer to this link : 
https://docs.strapi.io/dev-docs/api/rest/populate-select#population